pkg/cluster/defaults.go

```diff
@@ -28,3 +31,42 @@ func (d ControlPlaneIPCheckAnnotationDefaulter) ControlPlaneIPCheckDefault(ctx c
 
 	return spec, nil
 }
+
+// MachineHealthCheckDefaulter is the defaulter created to configure the machine health check timeouts.
+type MachineHealthCheckDefaulter struct {
+	nodeStartupTimeout      string
+	unhealthyMachineTimeout string
+}
+
+// NewMachineHealthCheckDefaulter allows to create a new MachineHealthCheckDefaulter.
+func NewMachineHealthCheckDefaulter(nodeStartupTimeout, unhealthyMachineTimeout string) MachineHealthCheckDefaulter {
+	return MachineHealthCheckDefaulter{
+		nodeStartupTimeout:      nodeStartupTimeout,
+		unhealthyMachineTimeout: unhealthyMachineTimeout,
+	}
+}
+
+// MachineHealthCheckDefault sets the defaults for machine health check timeouts.
+func (d MachineHealthCheckDefaulter) MachineHealthCheckDefault(ctx context.Context, spec *Spec) (*Spec, error) {
+	if spec.Config.Cluster.Spec.MachineHealthCheck != nil && len(spec.Config.Cluster.Spec.MachineHealthCheck.NodeStartupTimeout) != 0 && len(spec.Config.Cluster.Spec.MachineHealthCheck.UnhealthyMachineTimeout) != 0 {
+		return spec, nil
+	}
+
+	if spec.Config.Cluster.Spec.MachineHealthCheck == nil {
+		spec.Config.Cluster.Spec.MachineHealthCheck = &anywherev1.MachineHealthCheck{}
+	}
+
+	if len(spec.Cluster.Spec.MachineHealthCheck.NodeStartupTimeout) == 0 {
+		if spec.Cluster.Spec.DatacenterRef.Kind == anywherev1.TinkerbellDatacenterKind && d.nodeStartupTimeout == constants.DefaultNodeStartupTimeout.String() {
+			spec.Config.Cluster.Spec.MachineHealthCheck.NodeStartupTimeout = constants.DefaultTinkerbellNodeStartupTimeout.String()
+		} else {
+			spec.Config.Cluster.Spec.MachineHealthCheck.NodeStartupTimeout = d.nodeStartupTimeout
```

Comment:
So we only set default for Tinkerbell? What will this be for the other providers, 0?

Reply:
the default for others for node startup is 10m. that gets set for the `d.nodeStartupTimeout`

Reply:
In that case, any reason we aren't doing `d.nodeStartupTimeout = constants.DefaultTinkerbellNodeStartupTimeout.String()` and then not having an else condition below?

Reply:
The `d.nodeStartupTimeout` is set in the factory based on if the flag is set or not in cluster Options.